src/System.Net.Primitives/tests/UnitTests/CookieInternalTest.cs

```diff
@@ -68,5 +68,94 @@ public void CookieParserGetServer_SetCookieHeaderValue_Success(string cookieStri
             int expectedCookieCount = expectedStrings.Length >> 1;
             Assert.Equal(expectedCookieCount, cookieCount);
         }
+
+        // This assumes that the default cookie behavior is RFC 6265, which it
+        // is not ("Default = Rfc2109" in Cookie.cs:20)
+        // TODO: Will the Default behavior change to RFC 6265? If not, specify appropriate CookieVariant
```

Comment:
> Will the Default behavior change to RFC 6265?

I think that's the right direction. But we are not going to make it default for now - it may happen after implementing all of its new specs. /cc: @davidsh 

So for now, we should just introduce a new `CookieVariant.Rfc6265`, so that any user can opt-in.

Reply:
>I think that's the right direction. But we are not going to make it default for now - it may happen after implementing all of its new specs. /cc: @davidsh

In many ways RFC 6265 is a superset of the previous RFC's.  So, I think the right thing to do is to make the new "default" for .NET Core.  There are some cases that we will have to choose what we do.  The best path forward is to look at these cases and compare them to the top browsers and see how they handle this.  Most of the changes deal with cookie inclusion regarding sub paths.

Reply:
> In many ways RFC 6265 is a superset of the previous RFC's. So, I think the right thing to do is to make the new "default" for .NET Core.

Sounds good to me.

If we are changing the default, some old tests have to be modified accordingly, since they are no longer testing the "default" behavior.